iso15118/secc/states/iso15118_2_states.py

```diff
@@ -412,6 +412,24 @@ async def process_message(
 
         service_detail_req: ServiceDetailReq = msg.body.service_detail_req
 
+        is_found = False
+        for service_details in self.comm_session.offered_services:
+            if service_detail_req.service_id == service_details.service_id:
+                is_found = True
+                break
+
+        if not is_found:
+            # [V2G2-425] The SECC shall respond with a negative ResponseCode
+            # 'FAILED_ServiceIDInvalid'
+            # if the EVCC provided a not previously retrieved ServiceID in the
+            # ServiceDetailsReq.
+            self.stop_state_machine(
+                f"[V2G2-425] {service_detail_req.service_id} is not offered",
+                message,
+                ResponseCode.FAILED_SERVICE_ID_INVALID,
+            )
+            return
```

Comment:
This is no reason to stop the communication. we just need to communicate that that service ID is invalid and continue waiting for the next ServiceDetailReq or the next state

Reply:
Done